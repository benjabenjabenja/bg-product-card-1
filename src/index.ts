/**
 * ACA SE EXPORTAN TODOS LOS COMPONENTES, HOOKS, STYLES, Y LO QUE SEA QUE SE QUIERA EXPORTAR
 */

// COMPONENTES
export * from './components'

