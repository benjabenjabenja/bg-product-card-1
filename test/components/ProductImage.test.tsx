import React from 'react';
import { render } from '@testing-library/react';
import { ProductImage, ProductCard } from '../../src';
import { fakeMockProduct2 } from './data/products';

const product = fakeMockProduct2;

describe('ProductImage', () => {

    test('should render component image with custom image', () => {
        const { container } = render(<ProductImage img="https://test.com/image.jpg" />);
        expect(container).toMatchSnapshot();

        expect(container.querySelector('img')).toBeDefined();
        expect(container.querySelector('img')?.src).toBe('https://test.com/image.jpg');
    })

    test('should render component with image of product from product context', () => {
        const { container } = render(<ProductCard product={product}>
                {
                    () => <ProductImage />
                }
            </ProductCard>
        );

        expect(container).toMatchSnapshot();
    })

})

