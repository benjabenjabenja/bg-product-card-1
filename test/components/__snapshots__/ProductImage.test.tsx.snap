// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage should render component image with custom image 1`] = `
<div>
  <img
    alt="Product"
    class="productImg undefined"
    src="https://test.com/image.jpg"
  />
</div>
`;

exports[`ProductImage should render component with image of product from product context 1`] = `
<div>
  <div
    class="productCard undefined"
  >
    <img
      alt="Product"
      class="productImg undefined"
      src="./cofee-mug.png"
    />
  </div>
</div>
`;
